pkg/domain/filters/containers.go

```diff
@@ -146,10 +146,9 @@ func GenerateContainerFilterFuncs(filter string, filterValues []string, r *libpo
 			containerConfig := c.ConfigNoCopy()
 			var dest string
 			for _, filterValue := range filterValues {
-				arr := strings.SplitN(filterValue, ":", 2)
-				source := arr[0]
-				if len(arr) == 2 {
-					dest = arr[1]
+				source, newDest, hasNewDest := strings.Cut(filterValue, ":")
+				if hasNewDest {
+					dest = newDest
```

Comment:
this can be simplified, there is need for this condition. you can juts remove the dest declaration and do `srouce, dest, _ = strings.Cut()`
In fact the way the code is written now is broken because dest is declared outside of the loop which is wrong and causes incorrect matches
```
$ podman create -v /tmp/1:/1 -v /tmp/2:/2 alpine
pholzing ~ ✓ $ podman ps -a --filter volume=/doesnotexists:/3 --filter volume=/2
CONTAINER ID  IMAGE       COMMAND     CREATED     STATUS      PORTS       NAMES
pholzing ~ ✓ $ podman ps -a --filter volume=/2 --filter volume=/doesnotexists:/3
CONTAINER ID  IMAGE                            COMMAND     CREATED        STATUS      PORTS       NAMES
7a9a26f34eda  docker.io/library/alpine:latest  /bin/sh     4 minutes ago  Created                 jolly_keldysh
```

The first case is the bug, both cases should have matched because only one `volume` filter has to be true

Reply:
Updated to address bug